packages/editor/src/components/post-schedule/label.js

```diff
@@ -9,7 +9,7 @@ export function PostScheduleLabel( { date, isFloating } ) {
 	const settings = __experimentalGetSettings();
 
 	return date && ! isFloating ?
-		dateI18n( settings.formats.datetimeAbbreviated, date ) :
+		dateI18n( `${ settings.formats.date } ${ settings.formats.time }`, date ) :
```

Comment:
Hi @brentswisher, thank you for your contribution 👍 
Would it make sense to make this change in core:
https://github.com/WordPress/WordPress/blob/28a8f31ffa9dc238b56eb2413a728e861207b5af/wp-includes/script-loader.php#L615
Making datetimeAbbreviated format be the concatenation of time and date?

Reply:
Potentially, I wasn't sure where it was defined in core - so thanks for pointing that out!  That leads to another question - what is the purpose of the "datetimeAbbreviated" format? If I set my date and time format to custom things that are quite long, does it make sense to set something called dateTimeAbbreviated to that? Will updating that in core cause design issues in places where they are expecting a set formatting mask? There seemed to be a lot of questions I didn't know the answer to so I went with the smallest change to fix the issue, but if you think it warrants a larger discussion I am happy to pursue that as well.

Reply:
@jorgefilipecosta - My last comment had more questions and no answers so I tried to track this down a little better.

Investigating it looks like:
- 'datetimeAbbreviated' as a format was added in [this commit](https://github.com/WordPress/gutenberg/commit/00b9063c4b55d02f7fdfef2ea86757cd47a7a345) by @aldavigdis specifically for the use here in the post-schedule component
- It is not used anywhere else in Gutenberg I can find
- It is not used anywhere else in WordPress core I can find
- The "I can find" part here is important, I am not confident enough in my understanding of WordPress as a whole to make the call is really isn't used anywhere and can be removed

I would propose we:
1. Implement the change I have in this PR, so that 24hr time is supported and this issue can be closed.
2. Open a separate issue here and in trac asking if the 'datetimeAbbreviated' format needs to remain, as it looks like it is not used. That way it will hopefully get some more experienced eyes on it that may know the history before removing it.

Reply:
A lot of bike shedding went on when I worked on this a year ago, but `datetimeAbbreviated` was added to satisfy some design requirements and review if I remember it correctly.

I'm not sure if decisions had been made on how to handle localisation of dates by then, but at the time, there were some issues with parsing dates from PHP in the React components.

Reply:
Hi @swissspidy, would it be possible to provide some feedback regarding the i18n of this string?

Reply:
Both `settings.formats.date` and `settings.formats.time` are put through the translation function in the script loader. I've tested these changes with several different language settings and all looks to be working fine:
<img width="263" alt="Screen Shot 2019-08-16 at 7 08 22 pm" src="https://user-images.githubusercontent.com/8096000/63157040-6b84d300-c059-11e9-8f5a-23bff416e5a7.png">
I'm not sure what context `datetime` or `datetimeAbbreviated` could be useful in as they set a very specific formatting, as opposed to `date` and `time` which reflect the user settings. AFAICT neither of them are being used anywhere but, for the sake of fixing this annoying bug quickly, I vote we get this in and perhaps create a ticket to clean up all those formats later.